internal/rbd/controllerserver.go

```diff
@@ -1447,9 +1447,7 @@ func (cs *ControllerServer) ControllerExpandVolume(
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 	defer cr.DeleteCredentials()
-
-	rbdVol, err := GenVolFromVolID(ctx, volID, cr, req.GetSecrets())
-	defer rbdVol.Destroy()
```

Comment:
what is the reason for this change?

Reply:
madhu, just the parsing and volID creation is in the helper.

Reply:
This is not correct. the vol needs to be destroyed even in case of error.

Reply:
ah... you were referring to defer statement.. that can be brought back.